apps/backend/src/core/featureFlags/index.ts

```diff
@@ -0,0 +1,46 @@
+import { User } from '@auto-drive/models'
+import { config } from '../../config.js'
+import { createLogger } from '../../infrastructure/drivers/logger.js'
+
+const logger = createLogger('core:featureFlags')
+
+interface FeatureFlag {
+  active: boolean
+  employeeOnly?: boolean
+}
+
+const get = (user: User | null) => {
+  const entries = Object.entries(config.featureFlags.flags)
+
+  return Object.fromEntries(
+    entries.map(([key, value]) => [key, isActive(value, user)]),
+  )
+}
+
+const isActive = (value: FeatureFlag, user: User | null) => {
+  if (value.active) {
+    return true
+  }
+
+  return value.employeeOnly && isEmployee(user)
+}
+
+const isEmployee = (user: User | null) => {
+  logger.debug('Checking if user is employee:', user)
+  return Boolean(
+    user &&
+      user.oauthProvider !== 'web3-wallet' &&
+      config.featureFlags.employeeDomains
+        .filter((domain) => domain)
+        .some(
+          (domain) =>
+            user.oauthUsername &&
+            user.oauthUsername.toLowerCase().endsWith(domain),
```

Comment:
This check isn't quite correct.

For example, if the list was:
`autonomys.xyz`

Then `endsWith` would also match:
```
fakeautonomys.xyz
wwwautonomys.xyz
```

I think we'll want to do a domain-component aware match, so something like:
1. Split the username on `@` and `.`, and discard the user part (before the `@`)
2. Split the domain on `.`
3. Check the components of the username and domain, in reverse order, stopping when you get to the end of the domain

If we end up supporting full email addresses, we can check if the "domain" contains an `@`, and if it does, do a case-insensitive match of the full strings.

Reply:
I understand the issue, my intent with this check was to support as well the addition of complete emails. So, for adding a specific domain, the config should be: `@autonomys.xyz` rather than `autonomys.xyz`.

Reply:
I think that would be best.

If we have a separate domain list, then we can add `@` to the start of every domain in the list, if it doesn't start with it already. That would reject subdomains, which is probably ok.

And email lists should match the entire string.